### PR TITLE
DateTimePickerControl: Allow time to be set to beginning or end of day when in date-only mode

### PR DESCRIPTION
**Pull request by mattsherman**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Needed for #34538

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Run unit tests: `pnpm --filter=@woocommerce/components run test -- src/date-time-picker-control`
2. Interact with Storybook and make sure DateTimePickerControl "Controlled Date Only End Of Day" story returns the end of the day for the selected date: `pnpm --filter=@woocommerce/storybook storybook`
    - View the `onChange` entries in the Storybook "Actions" tab to confirm that the time is set to "23:59:59"

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx [hunk 1]
```diff
@@ -49,6 +50,7 @@ export const DateTimePickerControl: React.FC< DateTimePickerControlProps > = ( {
 	currentDate,
 	isDateOnlyPicker = false,
 	is12HourPicker = true,
+	timeForDateOnly,
```
Comment (louwie17) [inline packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx]: Should this default to `start-of-day`?

**Review by louwie17 — COMMENTED** [review]
@mattsherman this tested well and code looks good, I did leave one comment in relation to defaulting `timeForDateOnly` to `start-of-day`.

I also can't quite decide if I like this change or not, to me this logic seems better of to be dealt with by the consumer of the component, instead of adding the extra logic as prop options.

We could potentially expose some helper functions for this, but I guess that would in turn just be a wrapper for `momentjs`. 

In summary, would it be relatively simple to handle this in the consumers `onChange` function instead?
Or would this become to complicated without having `moment` readily available. If it's a one/two line thing in the consumer, I say leave it up to them. As we did only need this for end of day, given we already return it as a start of day.

File: packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx (hunk 1 above)
Comment (mattsherman) [reply packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx]: Good idea. I will make this change.

**Comment by mattsherman** [comment]
> I also can't quite decide if I like this change or not, to me this logic seems better of to be dealt with by the consumer of the component, instead of adding the extra logic as prop options.

I get that. I went back and forth on this initially but ultimately landed on it being something the component should/could easily handle due to the details below...

> We could potentially expose some helper functions for this, but I guess that would in turn just be a wrapper for `momentjs`.
> 
> In summary, would it be relatively simple to handle this in the consumers `onChange` function instead? Or would this become to complicated without having `moment` readily available. If it's a one/two line thing in the consumer, I say leave it up to them. As we did only need this for end of day, given we already return it as a start of day.

For most consumers, I believe that a start of day or end of day handling of the time when in date only mode will be most common. And, currently consumers don't have to do any datetime manipulation (which is often complex due to having to worry about timezones and such). So, I wanted to spare consumers of any complication for such a common and simple behavior. 

We already have the local datetime in the component, because that is how we are displaying it and getting it back from the picker. So, it is easy to add in this start/end manipulation.

**Comment by mattsherman** [comment]
Pulling this back In Progress to rebase and address comment.

**Comment by mattsherman** [comment]
@louwie17 I've rebased this and made the change to the default value for `timeForDateOnly`. It's ready for a re-review when you have the opportunity.

**Comment by louwie17** [comment]
> For most consumers, I believe that a start of day or end of day handling of the time when in date only mode will be most common. And, currently consumers don't have to do any datetime manipulation (which is often complex due to having to worry about timezones and such). So, I wanted to spare consumers of any complication for such a common and simple behavior.
> We already have the local datetime in the component, because that is how we are displaying it and getting it back from the picker. So, it is easy to add in this start/end manipulation.

Gotcha that makes sense, and I think you are right that these two use cases would be the most common.

**Review by louwie17 — APPROVED** [review]
Thanks for the rebase and addressing the PR feedback, this looks good and still tested well. :shipit:
